Lib/_osx_support.py

```diff
@@ -140,6 +140,20 @@ def _supports_universal_builds():
             osx_version = ''
     return bool(osx_version >= (10, 4)) if osx_version else False
 
+def _supports_arm64_builds():
+    """Returns True if arm64 builds are supported on this system"""
+    # There are two sets of systems supporting macOS/arm64 builds:
+    # 1. macOS 11 and later, unconditionally
+    # 2. macOS 10.15 with Xcode 12.2 or later
+    # For now the second category is ignored.
```

Comment:
I've discovered in another issue that official releases are build on old-ish macos.
Is this patch only for self-built python then?
Or is the intention to release arm and x86 packages separately?
Or perhaps to release "classic x86" and "arm+x86" packages?

Reply:
We're still working on this so we're not quite ready yet to say exactly what will be rolled out with which releases but our intent is to at least provide a universal installer variant that supports macOS 11.0 Big Sur and earlier systems and both Intel-64 and Apple Silicon natively.

Reply:
I don't have a definitive answer for your question, but my goal is to make it possible to build the official installers on a recent system.  That has several advantages: (1) it is required for arm64, (2) you can use new APIs when using a new system, (3) newer compilers generate better code.

I'd personally like to see 1 installer for Python 3.10: x86_64+arm64 for macOS 10.9 or later.  I don't know yet what the best path forward is for 3.9 (and possibly 3.8)

And current installers are indeed build on an old system, before this PR the only way to have an installer that works on macOS 10.9 and later is to build using the 10.9 SDK.

Reply:
From our perspective, the addition of runtime availability checks as part of this change is fantastic. We ship our own custom fork of Python in our macOS app, which we build using the standard/recommended approach: use the latest Xcode and macOS SDK + set your deployment target to the minimum version you want to support at runtime.